SystemInformer/syssccpu.c

```diff
@@ -1903,3 +1925,87 @@ PPH_STRINGREF PhGetHybridProcessorType(
     return NULL;
 }
 #endif
+
+/**
+ * \brief Checks if a specific logical processor (core) in the system is parked.
+ *
+ * This function determines whether a specific logical processor in the system
+ * is parked, meaning it is temporarily unavailable for the thread scheduler.
+ * It utilizes CPU sets functionality available starting with Windows 10.
+ *
+ * \param[in] ProcessorIndex The index of the logical processor to check.
+ *
+ * \return TRUE if the specified logical processor is parked, FALSE otherwise or
+ * in case of failure.
+ */
+BOOLEAN PhIsCoreParked(
+    _In_ ULONG ProcessorIndex
+    )
+{
+    PSYSTEM_CPU_SET_INFORMATION cpuSets = NULL;
+    ULONG returnedLength;
+    HANDLE currentProcess = GetCurrentProcess();
+
+    //
+    // To get core parking status of a specific logical processor in user mode,
+    // the CPU Sets functionality is used, which is available starting with Windows 10.
+    //
+    if (WindowsVersion < WINDOWS_10)
+    {
+        return FALSE;
+    }
+
+    //
+    // There is one CPU Set for each logical processor in the system.
+    //
+    if (ProcessorIndex > (ULONG)(PhSystemProcessorInformation.NumberOfProcessors - 1))
+    {
+        return FALSE;
+    }
+
+    //
+    // Pass NULL with a buffer length of 0 to get the required buffer size.
+    //
+    GetSystemCpuSetInformation(NULL,
+                               0,
+                               &returnedLength,
+                               currentProcess,
+                               0);
+
+    cpuSets = PhAllocate(returnedLength);
+    if (cpuSets == NULL)
+    {
+        return FALSE;
+    }
+
+    //
+    // Get system CPU Set information.
+    //
+    if (!GetSystemCpuSetInformation(cpuSets,
+                                    returnedLength,
+                                    &returnedLength,
+                                    currentProcess,
+                                    0))
+    {
+        goto CleanupExit;
+    }
+
+    //
+    // Check if the set of the target logical processor is parked.
+    //
+    if (cpuSets[ProcessorIndex].CpuSet.Parked) {
```

Comment:
`SYSTEM_CPU_SET_INFORMATION` is documented as variable-sized:
>This is a variable-sized structure designed for future expansion. When iterating over this structure, use the size field to determine the offset to the next structure.

Reply:
Nice observation, thanks!
Fixed: https://github.com/winsiderss/systeminformer/pull/2040/commits/5e510ad5bd74836c340c79b0bddf949f550ad1e6